### ISSUE TITLE
Improve the time taken by Github Actions on pull requests

### ISSUE THREAD

**Issue by VachaShah** [opensearch-project/opensearch-net#153]
Coming from https://github.com/opensearch-project/opensearch-clients/issues/53, update the workflows to reduce the amount of time taken by Github actions for integration tests.

**Comment by Yury-Fridlyand** [comment]
- [ ] Cache plugin build for existing integration tests (low impact, high benefit). See example: https://github.com/opensearch-project/opensearch-dsl-py/blob/2b9e688aa314098956fe7b7bd47618fc10340182/.github/workflows/integration-unreleased.yml#L31-L48
- [ ] Cache entire OpenSearch build (very low benefit)
- [ ] Use OpenSearch docker container (very low benefit)
- [ ] Run integration tests in parallel on all cluster types (low impact, high benefit)
- [ ] Add cache save-restore before and after cluster configuration procedure (high impact, low benefit) - need to do big changes in the test framework
- [ ] Add integration tests on unreleased versions and use cache as shown in p1

**Comment by VachaShah** [comment]
Looks like this is already done for this repo.
